determined-ai/determined
# feat: Experiment List comparison view should show hyperparameters chart [WEB-991]

## Description

The "compare" view now shows the hyperparameters parallel coordinates chart for each experiment. A few things to note:

1. We are not showing the "trials table" underneath the chart like we do elsewhere in the application. 
2. Only the experiments that have a value for **all** of the selected Hyperparameters and for the selected metric will be shown in the chart. 

**Notes about an issue with the user selectable "Scale"**:
While implementing this PR I came to the realization that the `log` argument no longer behaves as expected, it is clear from the backend code: https://github.com/determined-ai/determined/blob/main/master/internal/api_trials.go#L660

that the log argument no longer has any effect on the data returned from the `time-series` endpoint. The `logScale` argument previously was used to determine the behavior of downsampling and return the log value of metric data: https://github.com/determined-ai/determined/pull/6317/files#diff-b12c0f954792a2c30a89f05a1e7feee604255a0e4ed97569ec05786fd62d1bdbL57

however that is no longer the case and requesting both `linear` and `log` data will return the same result. I will create a bug ticket for the time being.



## Test Plan
1. Navigate to the new experiment listing page and click the "Compare" button
2. Without selecting any experiments, click the hyperparameters tab, no data should be shown.
3. Select an experiment, the "HP", and "Metric" dropdowns should be populated with values from the experiment, and the chart should show data. 
4. Select more experiments, the dropdowns should show the available metrics and hyperparameters from the new experiment salso. 
5. Choose HPs, and a Metric from the dropdown menu. Only experiments with values for all selected HPs and the selected metric should be shown. 
6. Ensure that color of the lines on the chart also match the fill color of the selected experiments. 



## Commentary (optional)

I filed the bug ticket: https://hpe-aiatscale.atlassian.net/browse/WEB-1338 to address the "scale" issue mentioned above.




## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

WEB-991
